homeassistant/components/google_generative_ai_conversation/config_flow.py

```diff
@@ -126,53 +119,84 @@ async def async_step_init(
     ) -> ConfigFlowResult:
         """Manage the options."""
         if user_input is not None:
-            return self.async_create_entry(
-                title="Google Generative AI Conversation", data=user_input
-            )
-        schema = google_generative_ai_config_option_schema(self.config_entry.options)
+            return self.async_create_entry(title="", data=user_input)
+        schema = await google_generative_ai_config_option_schema(
+            self.hass, self.config_entry.options
+        )
         return self.async_show_form(
             step_id="init",
             data_schema=vol.Schema(schema),
         )
 
 
-def google_generative_ai_config_option_schema(
+async def google_generative_ai_config_option_schema(
+    hass: HomeAssistant,
     options: MappingProxyType[str, Any],
 ) -> dict:
     """Return a schema for Google Generative AI completion options."""
-    if not options:
-        options = DEFAULT_OPTIONS
+    api_models = await hass.async_add_executor_job(partial(genai.list_models))
+
+    models: list[SelectOptionDict] = [
+        SelectOptionDict(
+            label="Gemini 1.0 Pro (free in US)",
```

Comment:
I'd remove the free in US. It's free and rated limited only if the linked cloud project doesn't have billing. It's currently the same for 1.5 but that will switch to paid starting May 30.